pkg/controller/endpointslice/endpointslice_controller.go

```diff
@@ -302,20 +307,35 @@ func (c *Controller) syncService(key string) error {
 	return nil
 }
 
-// obj could be a *v1.Service or a DeletionalFinalStateUnknown marker item
+// obj could be a *v1.Service.
 func (c *Controller) enqueueService(obj interface{}) {
+	if key, err := enqueueKey(obj, c.queue); err == nil {
+		_ = c.serviceSelectorCache.Update(key, obj.(*v1.Service))
```

Comment:
Since enqueuing the Service will result in a syncService call, it's important for the selectorCache to be updated first - I think the order needs to be switched here.